lib/gui/models/settings.js

```diff
@@ -34,12 +36,25 @@ const Store = require('./store');
  * settings.set('unmountOnSuccess', true);
  */
 exports.set = (key, value) => {
+  if (_.isNil(key)) {
+    throw errors.createError({
+      title: 'Missing setting key'
+    });
+  }
+
+  if (!_.isString(key)) {
+    throw errors.createError({
+      title: `Invalid setting key: ${key}`
+    });
+  }
+
+  const newSettings = _.assign(exports.getAll(), _.fromPairs([
+    [ key, value ]
```

Comment:
I don't know which is preferable but you can also do this, and remove the `fromPairs` usage.

```js
const newSettings = _.assign(exports.getAll(), {
  [key]: value
});
```